src/dispatch/case_cost/scheduled.py

```diff
@@ -0,0 +1,68 @@
+import logging
+
+from schedule import every
+
+from dispatch.database.core import SessionLocal
+from dispatch.decorators import scheduled_project_task, timer
+from dispatch.case import service as case_service
+from dispatch.case.enums import CaseStatus
+from dispatch.case_cost_type import service as case_cost_type_service
+from dispatch.project.models import Project
+from dispatch.scheduler import scheduler
+
+from .service import (
+    calculate_case_response_cost,
+    get_or_create_default_case_response_cost,
+)
+
+
+log = logging.getLogger(__name__)
+
+
+@scheduler.add(every(10).seconds, name="calculate-cases-response-cost")
+@timer
+@scheduled_project_task
+def calculate_cases_response_cost(db_session: SessionLocal, project: Project):
+    """Calculates and saves the response cost for all cases."""
+    response_cost_type = case_cost_type_service.get_default(
+        db_session=db_session, project_id=project.id
+    )
+
+    if response_cost_type is None:
+        log.warning(
+            f"A default cost type for response cost doesn't exist in the {project.name} project and organization {project.organization.name}. Response costs for cases won't be calculated."
+        )
+        return
+
+    cases = case_service.get_all(db_session=db_session, project_id=project.id)
+
+    for case in cases:
+        try:
+            # we get the response cost for the given case
+            case_response_cost = get_or_create_default_case_response_cost(case, db_session)
+
+            # we don't need to update the cost of closed cases if they already have a response cost and this was updated after the case was closed
+            if case.status == CaseStatus.closed:
+                if case_response_cost:
+                    if case_response_cost.updated_at > case.closed_at:
+                        continue
+            # we don't need to update the cost of escalated cases if they already have a response cost and this was updated after the case was escalated
+            if case.status == CaseStatus.escalated:
+                if case_response_cost:
+                    if case_response_cost.updated_at > case.escalated_at:
+                        continue
```

Comment:
Is there a way we can pull less cases here as opposed to all of them to optimize performance? I may be wrong here but couldn't we move these conditions for `closed` and `escalated` to the database query above instead of `get_all`?

Reply:
This is to cover cases that were marked `closed` or `escalated` in the time since the last cost update.

Reply:
Another optimization I think we can make here is selecting only the columns we need to compute the cost as it essentially does all of them currently, I believe.

Reply:
[Done](https://github.com/Netflix/dispatch/pull/4900/commits/be30c8d4501537ead81c22b992af05f8afd2fe91#diff-85562e526dc8615124aead4d1b1dc02d9fbfdbc211396e4c880f3e63f354fb00R6)

Reply:
Should we need to use one of these service functions (`get_all_by_status`) instead of `get_all` above?

Reply:
Good point. [Fixed it here](https://github.com/Netflix/dispatch/pull/4900/commits/128a9db2628764a60122d360efc79fd561cab221).

Reply:
The function call is missing the `status` parameter and my guess is we'll need to pull multiple different status so may need an update to accept a list of statuses.

Reply:
Done.